sklearn/feature_selection/tests/test_sequential.py

```diff
@@ -23,13 +24,15 @@ def test_bad_n_features_to_select(n_features_to_select):
 
 def test_bad_direction():
     X, y = make_regression(n_features=5)
-    sfs = SequentialFeatureSelector(LinearRegression(), direction="bad")
+    sfs = SequentialFeatureSelector(
+        LinearRegression(), n_features_to_select="auto", direction="bad"
+    )
     with pytest.raises(ValueError, match="must be either 'forward' or"):
         sfs.fit(X, y)
 
 
 @pytest.mark.parametrize("direction", ("forward", "backward"))
-@pytest.mark.parametrize("n_features_to_select", (1, 5, 9, None))
+@pytest.mark.parametrize("n_features_to_select", (1, 5, 9))
```

Comment:
I think we can add "auto" here to test the `tol=None` + `"auto"` case. This way we can simplify the `test_n_features_to_select_auto` to test check for `tol=1e-3` and `'auto'`.

Reply:
simplify test with https://github.com/scikit-learn/scikit-learn/pull/20145/commits/649fc31e9725970ab82845de8b4774bc1894472f

I'm not confident in this fix.
Does this fix meet @thomasjpfan needs?